src/metabase/query_processor/parameters.clj

```diff
@@ -30,83 +128,83 @@
                                                                    "Q3" DateTimeConstants/JULY
                                                                    "Q4" DateTimeConstants/OCTOBER))))
 
-(defn- week-range [^DateTime dt]
-  ;; weeks always start on SUNDAY and end on SATURDAY
-  ;; NOTE: in Joda the week starts on Monday and ends on Sunday, so to get the right Sunday we rollback 1 week
-  {:end   (.withDayOfWeek dt DateTimeConstants/SATURDAY)
-   :start (.withDayOfWeek ^DateTime (t/minus dt (t/weeks 1)) DateTimeConstants/SUNDAY)})
+;; NOTE: this is perhaps a little hacky, but we are assuming that `dt` will be in the first month of the quarter
+(defn- quarter-range
+  [quarter year]
+  (let [dt (start-of-quarter quarter year)]
+    {:end   (t/last-day-of-the-month (t/plus dt (t/months 2)))
+     :start (t/first-day-of-the-month dt)}))
 
-(defn- month-range [^DateTime dt]
-  {:end   (t/last-day-of-the-month dt)
-   :start (t/first-day-of-the-month dt)})
+(def ^:private absolute-date-param-values
+  ;; year and month
+   [{:parser (regex->parser #"([0-9]{4}-[0-9]{2})" [:date])
+    :range  (fn [{:keys [date]} _]
+                (month-range date date))
+    :filter (fn [{:keys [date]} field]
+              (range->filter (month-range date date) field))}
+  ;; quarter year
+   {:parser (regex->parser #"(Q[1-4]{1})-([0-9]{4})" [:quarter :year])
+    :range (fn [{:keys [quarter year]} _]
+               (quarter-range quarter (Integer/parseInt year)))
+    :filter (fn [{:keys [quarter year]} field]
+               (range->filter (quarter-range quarter (Integer/parseInt year))
+                              field))}
+  ;; single day
+  {:parser (regex->parser #"([0-9-T:]+)" [:date])
+    :range  (fn [{:keys [date]} _]
+              {:start date, :end date})
+    :filter (fn [{:keys [date]} field]
+              (let [iso8601date (date->iso8601 date)]
+                ["BETWEEN" field iso8601date iso8601date]))}
+   ;; day range
+   {:parser (regex->parser #"([0-9-T:]+)~([0-9-T:]+)" [:date-1 :date-2])
+    :range  (fn [{:keys [date-1 date-2]} _]
+              {:start date-1, :end date-2})
+    :filter (fn [{:keys [date-1 date-2]} field]
+              ["BETWEEN" field (date->iso8601 date-1) (date->iso8601 date-2)])}
+   ;; before day
+   {:parser (regex->parser #"~([0-9-T:]+)" [:date])
+    :range  (fn [{:keys [date]} _]
+              {:end date})
+    :filter (fn [{:keys [date]} field]
+              ["<" field (date->iso8601 date)])}
+   ;; after day
+   {:parser (regex->parser #"([0-9-T:]+)~" [:date])
+    :range  (fn [{:keys [date]} _]
+              {:start date})
+    :filter (fn [{:keys [date]} field]
+              [">" field (date->iso8601 date)])}
+])
```

Comment:
Closing brackets/parens (`])`) don't get their own lines in Clojure.

This should be 

```clojure
[">" field (date->iso8601 date)])}])
```